kratos/geometries/point.h

```diff
@@ -195,16 +197,138 @@ class Point : public array_1d<double, 3>
         return this->operator[](2);
     }
 
+    /**
+     * @brief This method gets the current coordinates (constant)
+     * @return The current coordinates
+     */
     CoordinatesArrayType const &Coordinates() const
     {
         return *this;
     }
 
+    /**
+     * @brief This method gets the current coordinates (reference)
+     * @return The current coordinates
+     */
     CoordinatesArrayType &Coordinates()
     {
         return *this;
     }
 
+    /**
+     * @brief This method gets the initial coordinates (constant)
+     * @return The initial coordinates
+     * @note Must be override in tehd erived classes with initial position
+     */
+    virtual const PointType& GetInitialPosition() const
+    {
+        KRATOS_WARNING("Point") << "This class does not have InitialPosition coordinates. Use an derived class and override this method in order to do so" << std::endl;
+        return *this;
+    }
+
+    /**
+     * @brief This method gets the initial coordinates (reference)
+     * @return The initial coordinates
+     * @note Must be override in tehd erived classes with initial position
+     */
+    virtual PointType& GetInitialPosition()
+    {
+        KRATOS_WARNING("Point") << "This class does not have InitialPosition coordinates. Use an derived class and override this method in order to do so" << std::endl;
+        return *this;
+    }
+
+    /**
+     * @brief This method returns the X initial coordinates (reference)
+     * @return X initial coordinates
+     * @note Must be override in tehd erived classes with initial position
+     */
+    virtual double& X0()
```

Comment:
how big is the impact of making this virtual?
We recently saw that this had quite an impact when testing it for the ShapFunctions

Reply:
Huuum, I don't know how to measure it properly. Do you suggest a similar approach? 

https://stackoverflow.com/questions/667634/what-is-the-performance-cost-of-having-a-virtual-method-in-a-c-class

Reply:
What @tteschemacher did was to create a dummy test measuring the elapsed time of calling the method thousands (or millions) of times with and without the `virtual` clause. I think this could do the job in here as well.

Reply:
OK, I will see